### PR TITLE
sol Passing arguments to the base constructor.

### PR DESCRIPTION
**Pull request by LianaHus**
the story is here: https://www.pivotaltracker.com/n/projects/1189488/stories/88454388

### CODE REVIEW

File: libsolidity/CompilerContext.cpp [hunk 1]
```diff
@@ -122,6 +123,16 @@ eth::AssemblyItem CompilerContext::getSuperFunctionEntryLabel(string const& _nam
 	return m_asm.newTag(); // not reached
 }
 
+FunctionDefinition const* CompilerContext::getNextConstructor(ContractDefinition const& _contract) const
+{
+	vector<ContractDefinition const*>::const_iterator it = getSuperContract(_contract);
+	for (; it != m_inheritanceHierarchy.end(); it = getSuperContract(**it))
```
Comment (chriseth) [inline libsolidity/CompilerContext.cpp]: `++it` should be enough (not `it = getSuperContract(**it)`).

File: libsolidity/AST.cpp [hunk 2]
```diff
@@ -324,19 +326,34 @@ void ModifierDefinition::checkTypeRequirements()
 	m_body->checkTypeRequirements();
 }
 
-void ModifierInvocation::checkTypeRequirements()
+void ModifierInvocation::checkTypeRequirements(std::vector<ASTPointer<InheritanceSpecifier>> const& _bases)
```
Comment (chriseth) [inline libsolidity/AST.cpp]: `std::` not needed.

File: libsolidity/AST.cpp [hunk 3]
```diff
@@ -324,19 +326,34 @@ void ModifierDefinition::checkTypeRequirements()
...
-	solAssert(modifier, "Function modifier not found.");
-	vector<ASTPointer<VariableDeclaration>> const& parameters = modifier->getParameters();
-	if (parameters.size() != m_arguments.size())
+	auto declaration = m_modifierName->getReferencedDeclaration();
+	vector<ASTPointer<VariableDeclaration>> emptyParameterList;
+	vector<ASTPointer<VariableDeclaration>> const* parameters = nullptr;
+	if (auto modifier = dynamic_cast<ModifierDefinition const*>(declaration))
+		parameters = &modifier->getParameters();
+	else
+		for (auto const& base: _bases)
```
Comment (chriseth) [inline libsolidity/AST.cpp]: Some comments about what is done here?

File: libsolidity/AST.h [hunk 4]
```diff
@@ -516,11 +516,12 @@ class ModifierInvocation: public ASTNode
 	ASTPointer<Identifier> const& getName() const { return m_modifierName; }
 	std::vector<ASTPointer<Expression>> const& getArguments() const { return m_arguments; }
 
-	void checkTypeRequirements();
+	void checkTypeRequirements(std::vector<ASTPointer<InheritanceSpecifier>> const& _bases);
```
Comment (chriseth) [inline libsolidity/AST.h]: Documentation about the parameter

File: libsolidity/AST.cpp (hunk 3 above)
Comment (LianaHus) [inline libsolidity/AST.cpp]: sure. there are some more "std::"'s in this file, should I remove them all?(not from new code)

File: libsolidity/AST.cpp (hunk 2 above)
Comment (LianaHus) [inline libsolidity/AST.cpp]: sure. there are some more "std::"'s in this file, should I remove them all?(not from new code)

**Comment by LianaHus** [comment]
rebased

**Pull request merged**